pkg/scheduler/cache/interface.go

```diff
@@ -91,14 +80,9 @@ type Cache interface {
 
 	// EventRecorder returns the event recorder
 	EventRecorder() record.EventRecorder
-}
 
-// VolumeBinder interface for allocate and bind volumes
-type VolumeBinder interface {
-	GetPodVolumes(task *api.TaskInfo, node *v1.Node) (*volumebinding.PodVolumes, error)
-	RevertVolumes(task *api.TaskInfo, podVolumes *volumebinding.PodVolumes)
-	AllocateVolumes(task *api.TaskInfo, hostname string, podVolumes *volumebinding.PodVolumes) error
-	BindVolumes(task *api.TaskInfo, podVolumes *volumebinding.PodVolumes) error
+	// RegisterPreBinder registers the passed preBinder to the cache's preBinderRegistry
+	RegisterPreBinder(name string, preBinder interface{})
```

Comment:
How about use a common register, because maybe we will add other extension besides Prebind.

Reply:
And if not, every time there is a new extension we need add a new interface.

Reply:
OK I have renamed it to RegisterBinder, if the passed in binder can be asserted as a PreBinder or PostBinder, it can be added into the BinderRegistry's prebinder map or postbinder map:
```go
// BinderRegistry is used to hold the registered binders, such as pre-binders, post-binders
type BinderRegistry struct {
	mu         sync.RWMutex
	preBinders map[string]PreBinder
	// Can add postBinders in the future
}
```